ecs_pod.go

```diff
@@ -20,17 +22,67 @@ type ECSPod interface {
 
 // BasicECSPod represents a pod that is backed by ECS.
 type BasicECSPod struct {
-	opts BasicECSPodOptions
+	client    ECSClient
+	vault     secret.Vault
+	resources ECSPodResources
+	status    ECSPodStatus
 }
 
-// BasicECSPodOptions are options to create a BasicECSPod.
+// BasicECSPodOptions are options to create a basic ECS pod.
 type BasicECSPodOptions struct {
 	Client    ECSClient
 	Vault     secret.Vault
 	Resources *ECSPodResources
 	Status    *ECSPodStatus
 }
 
+// NewBasicECSPodOptions returns new uninitialized options to create a basic ECS
+// pod.
+func NewBasicECSPodOptions() *BasicECSPodOptions {
+	return &BasicECSPodOptions{}
+}
+
+// SetClient sets the client the pod uses to communicate with ECS.
+func (o *BasicECSPodOptions) SetClient(c ECSClient) *BasicECSPodOptions {
+	o.Client = c
+	return o
+}
+
+// SetVault sets the vault that the pod uses to manage secrets.
+func (o *BasicECSPodOptions) SetVault(v secret.Vault) *BasicECSPodOptions {
+	o.Vault = v
+	return o
+}
+
+// SetResources sets the resources used by the pod.
+func (o *BasicECSPodOptions) SetResources(res ECSPodResources) *BasicECSPodOptions {
+	o.Resources = &res
+	return o
+}
+
+// SetStatus sets the current status for the pod.
+func (o *BasicECSPodOptions) SetStatus(s ECSPodStatus) *BasicECSPodOptions {
+	o.Status = &s
+	return o
+}
+
+// Validate checks that the required parameters to initialize a pod are given.
+func (o *BasicECSPodOptions) Validate() error {
+	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(o.Client == nil, "must specify a client")
+	catcher.NewWhen(o.Resources == nil, "must specify at least one underlying resource being used by the pod")
+	catcher.NewWhen(o.Resources != nil && o.Resources.TaskID == nil, "must specify task ID")
+	catcher.NewWhen(o.Status == nil, "must specify a status")
+	if o.Status != nil {
```

Comment:
Can we pull this out and create a validation func on `ECSPodStatus`?

Reply:
Good idea. Done!